XamarinCommunityToolkit/Views/RangeSlider.shared.cs

```diff
@@ -0,0 +1,625 @@
+            labelMaxHeight = maxHeight;
+            OnLayoutPropertyChanged();
+        }
+
+        void OnPanUpdated(object sender, PanUpdatedEventArgs e)
+        {
+            var view = (View)sender;
+            switch (e.StatusType)
+            {
+                case GestureStatus.Started:
+                    OnPanStarted(view);
+                    break;
+                case GestureStatus.Running:
+                    OnPanRunning(view, e.TotalX);
+                    break;
+                case GestureStatus.Completed:
+                    OnPanCompleted(view);
+                    break;
+                case GestureStatus.Canceled:
+                    OnPanCanceled(view);
+                    break;
+            }
+        }
+
+        void OnPanStarted(View view)
+            => thumbPositionMap[view] = view.TranslationX;
+
+        void OnPanRunning(View view, double value)
+            => UpdateValue(view, value + GetPanShiftValue(view));
+
+        void OnPanCompleted(View view)
+            => thumbPositionMap[view] = view.TranslationX;
+
+        void OnPanCanceled(View view)
+            => UpdateValue(view, thumbPositionMap[view]);
+
+        void UpdateValue(View view, double value)
+        {
+            var rangeValue = MaximumValue - MinimumValue;
+            if (view == LowerThumb)
+            {
+                LowerValue = Min(Max(MinimumValue, value / TrackWidth * rangeValue + MinimumValue), UpperValue);
+                return;
+            }
+            UpperValue = Min(Max(LowerValue, (value - LowerThumb.Width) / TrackWidth * rangeValue + MinimumValue), MaximumValue);
+        }
+
+        double GetPanShiftValue(View view)
+            => Device.RuntimePlatform == Device.Android
+                ? view.TranslationX
+                : thumbPositionMap[view];
+
+        void SetValueLabelBinding(Label label, BindableProperty bindableProperty)
+            => label.SetBinding(Label.TextProperty, new Binding
+            {
+                Path = bindableProperty.PropertyName,
+                Source = this,
+                StringFormat = ValueLabelStringFormat
+            });
+
+        void AddGestureRecognizer(View view, PanGestureRecognizer gestureRecognizer)
+        {
+            gestureRecognizer.PanUpdated += OnPanUpdated;
+            view.GestureRecognizers.Add(gestureRecognizer);
+        }
+
+        void RemoveGestureRecognizer(View view, PanGestureRecognizer gestureRecognizer)
+        {
+            gestureRecognizer.PanUpdated -= OnPanUpdated;
+            view.GestureRecognizers.Remove(gestureRecognizer);
+        }
+
+        Color GetColorOrDefault(Color color, Color defaultColor)
+            => color == Color.Default
+                ? defaultColor
+                : color;
+
+        double GetDoubleOrDefault(double value, double defaultSize)
+            => value < 0
+                ? defaultSize
+                : value;
+
+        sealed class RangeSliderLayout : AbsoluteLayout
```

Comment:
Should we move this into its own file maybe? Have a folder per control in the project and group files in there? This class isn't too big though. But if we get more controls like these that might help readability and discoverability

Reply:
Mmm, I don't think so. This layout doesn't make any sense without RangeSlider. And it even exists only because that's only one way I found how we can set content to TemplatedView.


This is how I worked around it for Expander in XF 
https://github.com/xamarin/Xamarin.Forms/blob/719fc7a604ff0cce8922d717c99bfb0fa17e35e0/Xamarin.Forms.Core/Expander.cs#L59

but InternalChildren collection is internal indeed :) So we cannot put control there directly, that is why I created this layout and put inside RangeSlider as a private class.

Reply:
Hmmm, just realized that child is added as soon as we register control template 
```ControlTemplate = new ControlTemplate(typeof(RangeSliderLayout));``

So probably we can remove this internal class at all

Reply:
@jfversluis got rid of useless SliderLayout